Tests/SonarScanner.MSBuild.Shim.Test/SonarScannerWrapperTests.cs

```diff
@@ -88,82 +88,69 @@ public void Ctor_WhenOperatingSystemProviderIsNull_Throws()
+        var wrapper = new SonarScannerWrapperTestRunner();
 
-        var success = ExecuteJavaRunnerIgnoringAsserts(config, EmptyPropertyProvider.Instance, testLogger, "c:\\exePath", "f:\\props.txt", mockRunner);
+        var result = wrapper.ExecuteJavaRunnerIgnoringAsserts();
 
-        VerifyProcessRunOutcome(mockRunner, testLogger, "c:\\workingDir", success, true);
-        testLogger.AssertInfoMessageExists(Resources.MSG_SonarScannerHomeIsSet);
+        result.VerifyProcessRunOutcome("C:\\working\\dir", true);
+        result.Logger.AssertInfoMessageExists(Resources.MSG_SonarScannerHomeIsSet);
     }
 
     [TestMethod]
     public void SonarScanner_StandardAdditionalArgumentsPassed()
     {
-        var logger = new TestLogger();
-        var mockRunner = new MockProcessRunner(executeResult: true);
-        var config = new AnalysisConfig { SonarScannerWorkingDirectory = "c:\\work" };
-
-        var success = ExecuteJavaRunnerIgnoringAsserts(config, EmptyPropertyProvider.Instance, logger, "c:\\exe.Path", "d:\\propertiesFile.Path", mockRunner);
+        var wrapper = new SonarScannerWrapperTestRunner();
+        var result = wrapper.ExecuteJavaRunnerIgnoringAsserts();
 
-        VerifyProcessRunOutcome(mockRunner, logger, "c:\\work", success, true);
+        result.VerifyProcessRunOutcome("C:\\working\\dir", true);
     }
 
     [TestMethod]
     public void SonarScanner_CmdLineArgsArePassedThroughToTheWrapperAndAppearFirst()
     {
-        var logger = new TestLogger();
-        var userArgs = new ListPropertiesProvider();
-        userArgs.AddProperty("sonar.login", "me");
-        userArgs.AddProperty("sonar.password", "my.pwd");
-        userArgs.AddProperty("sonar.token", "token");
-        var mockRunner = new MockProcessRunner(executeResult: true);
-
-        var success = ExecuteJavaRunnerIgnoringAsserts(
-            new AnalysisConfig { SonarScannerWorkingDirectory = "D:\\dummyWorkingDirectory" },
-            userArgs,
-            logger,
-            "c:\\dummy.exe",
-            "c:\\foo.properties",
-            mockRunner);
-
-        VerifyProcessRunOutcome(mockRunner, logger, "D:\\dummyWorkingDirectory", success, true);
-        CheckStandardArgsPassed(mockRunner, "c:\\foo.properties");
-        var loginIndex = CheckArgExists("-Dsonar.login=me", mockRunner);
-        var passwordIndex = CheckArgExists("-Dsonar.password=my.pwd", mockRunner);
-        var tokenIndex = CheckArgExists("-Dsonar.token=token", mockRunner);
-        var propertiesFileIndex = CheckArgExists(SonarScannerWrapper.ProjectSettingsFileArgName, mockRunner);
+        var wrapper = new SonarScannerWrapperTestRunner()
+        {
+            Config = { SonarScannerWorkingDirectory = "D:\\dummyWorkingDirectory" },
+            UserCmdLineArguments =
+            {
+                { "sonar.login", "me" },
+                { "sonar.password", "my.pwd" },
+                { "sonar.token", "token" }
+            },
```

Comment:
The changes in `ListPropertiesProvider` enable this dictionary initializer syntax, which I think is much nicer than the previous
```cs
var userArgs = new ListPropertiesProvider();
userArgs.AddProperty("sonar.login", "me");
userArgs.AddProperty("sonar.password", "my.pwd");
userArgs.AddProperty("sonar.token", "token");
```